homeassistant/components/utility_meter/__init__.py

```diff
@@ -71,6 +72,7 @@ def max_28_days(config):
 
 METER_CONFIG_SCHEMA = vol.Schema(
     vol.All(
+        cv.deprecated(CONF_NAME),
```

Comment:
This is odd behavior and has nothing to do with the addition of the unique ID. Please revert this.

Reply:
The point is that adding `unique_id` replaces the need to have the `name` option which was used for setting the `friendly_name` only.

Reply:
I get the point, but it's not a normal pattern and breaking for existing installations.
Also, there is no need to remove it in this PR either as it's not related.

Therefore, please revert this change.

Reply:
IMHO it will confuse users to set a friently_name in YAML and yet be able to change the friendly_name in UI / Customization (and these two always overwrite this `name` option)

Reply:
It is not a common pattern, none of our other helpers do this, therefore, lets revert it.

Additionally, it has nothing to do with the unique ID, so can be removed from this PR anyways (out of scope).